File: 백준/Silver/10845. 큐/큐.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int len = Integer.parseInt(br.readLine());
        int[] que = new int[10000];
        int front = 0, back = 0;
        for(int i=0; i<len; i++){
            String order = br.readLine();
            if(back >=0 && order.contains("push")){
                String[] spl = order.split(" ");
                que[back++] = Integer.parseInt(spl[1]);
            }else if(back>=front && order.equals("pop")){
                if(back==front){
                    sb.append(-1).append('\n');
                }else{
                    sb.append(que[front++]).append('\n');
                }
            }else if(order.equals("size")){
                if(back-front<0)
                    sb.append(0).append('\n');
                else
                    sb.append(back-front).append('\n');
            }else if(order.equals("empty")){
                if(back-front<=0)
                    sb.append(1).append('\n');
                else
                    sb.append(0).append('\n');
            }else if(order.equals("front")){
                if(back-front<=0)
                    sb.append(-1).append('\n');
                else
                    sb.append(que[front]).append('\n');
            }else if(order.equals("back")){
                if(back-front<=0)
                    sb.append(-1).append('\n');
                else
                    sb.append(que[back-1]).append('\n');
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/10845. 큐/큐.js
let [ N,...input ] = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n").map(v => v.split(" "));

var queue = [];
let answer = [];

for(var i=0; i<N; i++){
  var order = input[i];
  if(order[0].includes('push')){
    queue.push(order[1]);
  }
  else if(order[0].includes('pop')){
    answer.push(queue.shift() || -1);
  }
  else if(order[0].includes('size')){
    answer.push(queue.length);
  }
  else if(order[0].includes('empty')){
    answer.push(queue.length? 0 : 1);
  }
  else if(order[0].includes('front')){
    answer.push(queue[0] || -1);
  }
  else if(order[0].includes('back')){
    answer.push(queue[queue.length-1] ||-1);
  }
}
console.log(answer.join("\n"));